fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/MapPipelinedCursor.java

```diff
@@ -85,59 +80,18 @@ public MapPipelinedCursor(@Nonnull RecordCursor<T> inner, @Nonnull Function<T, C
         if (nextResult != null && !nextResult.hasNext()) {
             return CompletableFuture.completedFuture(nextResult);
         }
-        mayGetContinuation = false;
         return AsyncUtil.whileTrue(this::tryToFillPipeline, getExecutor())
                 // pipeline will necessarily contain something if we stopped looping, so pipeline.remove() is nonnull
                 .thenCompose(vignore -> pipeline.peek()) // future should already be (nearly) ready if we stopped looping
                 .thenApply(result -> {
                     if (result.hasNext()) {
                         pipeline.remove();
                     }
-                    mayGetContinuation = !result.hasNext();
                     nextResult = result;
                     return result;
                 });
     }
 
-    @Nonnull
-    @Override
-    @Deprecated
-    public CompletableFuture<Boolean> onHasNext() {
-        if (nextFuture == null) {
```

Comment:
Done. I also removed `innerFuture`, which apparently has been unused since #213, when its job was replaced by `waitInnerFuture`, which still is, in fact, being used. I looked over that usage, and it looks like leaving around `innerFuture` was just an oversight, so it should be fine to remove it now.